assets/src/edit-story/components/panels/design/preset/colorPreset/colorPresetActions.js

```diff
@@ -70,35 +129,109 @@ function ColorPresetActions({ color }) {
     (toAdd) => {
       if (toAdd) {
         // If match found, don't add.
-        // @todo UX improvement: notify the user/mark the existing color?
-        if (findMatchingColor(toAdd, globalStoryStyles, isText)) {
+        const currentStyles = showLocalColors
+          ? currentStoryStyles
+          : globalStoryStyles;
+        if (findMatchingColor(toAdd, currentStyles, isText)) {
           return;
         }
-
+        const newProps = showLocalColors
+          ? {
+              currentStoryStyles: {
+                ...currentStoryStyles,
+                colors: [...localColors, toAdd],
+              },
+            }
+          : {
+              globalStoryStyles: {
+                ...globalStoryStyles,
+                colors: [...globalColors, toAdd],
+              },
+            };
         updateStory({
-          properties: {
-            globalStoryStyles: {
-              ...globalStoryStyles,
-              colors: [...colors, toAdd],
-            },
-          },
+          properties: newProps,
         });
       }
     },
-    [globalStoryStyles, isText, colors, updateStory]
+    [
+      currentStoryStyles,
+      globalStoryStyles,
+      localColors,
+      showLocalColors,
+      isText,
+      globalColors,
+      updateStory,
+    ]
   );
 
+  const options = [
+    {
+      id: LOCAL,
+      name: __('Current colors', 'web-stories'),
+    },
+    {
+      id: GLOBAL,
+      name: __('Global colors', 'web-stories'),
```

Comment:
In the design, these are labeled "Current story" and "All stories" - I think that's a simpler language, that's probably better for this.

Reply:
Oops, not intentional.